patches/chrome-android-java-src-org-chromium-chrome-browser-undo_tab_close_snackbar-UndoBarController.java.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java b/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java
+index b0367689a7a34a3cfc39d7c16433df338a7f23bb..939198410e3f56b1ff67b3c12f0016de1c983448 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/undo_tab_close_snackbar/UndoBarController.java
+@@ -154,7 +154,7 @@ public class UndoBarController implements SnackbarManager.SnackbarController {
+      * @param closedTabs A list of tabs that were closed.
+      * @param isAllTabs Whether all tabs were closed.
+      */
+-    private void showUndoBar(List<Tab> closedTabs, boolean isAllTabs) {
++    protected void showUndoBar(List<Tab> closedTabs, boolean isAllTabs) {
```

Comment:
the patch isn't needed at all. We can make that method as private and Override it in `BraveUndoBarController`. Check that class as example https://github.com/brave/brave-core/blob/1.69.x/build/android/bytecode/java/org/brave/bytecode/BraveAppMenuClassAdapter.java#L22

Reply:
Add a test for method existence once you have the logic to make the method `public` and the `Override` annotation is applied to our class, see https://github.com/brave/brave-core/blob/1.69.x/android/javatests/org/chromium/chrome/browser/BytecodeTest.java#L337

Reply:
Thanks! I didn't know that.

Btw, Could you tell me how to run the test in Local?

Reply:
`npm run test -- brave_unit_tests Debug --target_os=android --target_arch=x86`, `Debug` is a configuration you build with, `x86` is arch

Reply:
Hey @SergeyZhukovsky! 

I tried running the test using the command above, but I'm encountering a Timeout Error, as shown below:

```
C 18:17:15.602   30.145s start_emulator_instance(android_30_google_atd_x86|128652590926240)  Timed out. Dumping threads.
C 18:17:15.602   30.145s start_emulator_instance(android_30_google_atd_x86|128652590926240)  ********************************************************************************
C 18:17:15.602   30.145s start_emulator_instance(android_30_google_atd_x86|128652590926240)  Stack dump for thread 'TimeoutThread-1-for-start_emulator_instance(android_30_google_atd_x86|128652590926240)'
C 18:17:15.602   30.145s start_emulator_instance(android_30_google_atd_x86|128652590926240)  **************************

....
```

I also tried installing the required AVD using the command `tools/android/avd/avd.py install --avd-config ...`, but still no luck. Do you have any ideas on how to resolve this?

Note: I've already done the prerequisites from [Chromium Android Emulator Setup](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/android_emulator.md)